src/test/java/co/cask/wrangler/steps/PipelineTest.java

```diff
@@ -79,21 +80,61 @@ public void testSplit() throws Exception {
     Assert.assertEquals("2,a,A", row.getValue("secondCol"));
   }
 
+  @Ignore
   @Test
   public void testSplitWithNull() throws Exception {
-    List<Step> steps = new ArrayList<>();
     Row row = new Row("col", "1,2,a,A");
 
     // Define all the steps in the wrangler.
-    steps.add(new Split(0,"","col","|","firstCol","secondCol"));
+    Step step = new Split(0,"","col","|","firstCol","secondCol");
 
     // Run through the wrangling steps.
-    for (Step step : steps) {
-      row = (Row) step.execute(row);
-    }
+    Row actual = (Row) step.execute(row);
+
+    Assert.assertEquals("1,2,a,A", actual.getValue("firstCol"));
+    Assert.assertNull(actual.getValue("secondCol"));
+  }
+
+  @Test
+  public void testMaskingSubstitution() throws Exception {
+    // Check valid status.
+    Row row = new Row("ssn", "888990000");
+    Step step = new Mask(0, "", "ssn", "xxx-xx-####", 1);
+    Row actual = (Row) step.execute(row);
+    Assert.assertEquals("xxx-xx-0000", actual.getValue("ssn"));
+    step = new Mask(0, "", "ssn", "x-####", 1);
+    actual = (Row) step.execute(row);
+    Assert.assertEquals("x-8899", actual.getValue("ssn"));
+  }
+
+  @Test
+  public void testMaskSuffle() throws Exception {
+    Row row = new Row("address", "150 Mars Street, Mar City, MAR, 783735");
+    Step step = new Mask(0, "", "address", "", 2);
+    Row actual = (Row) step.execute(row);
+    Assert.assertEquals("089 Kyrp Czsyyr, Dyg Goci, FAG, 720322", actual.getValue("address"));
```

Comment:
Seed is the same! So when you have the same seed, the random sequence is the same. It's based on linear congruential random numbers.